app/code/Magento/Sales/Model/EmailSenderHandler.php

```diff
@@ -66,8 +66,10 @@ public function __construct(
     public function sendEmails()
     {
         if ($this->globalConfig->getValue('sales_email/general/async_sending')) {
+            $this->entityCollection->clear();
             $this->entityCollection->addFieldToFilter('send_email', ['eq' => 1]);
             $this->entityCollection->addFieldToFilter('email_sent', ['null' => true]);
+            $this->entityCollection->setPageSize(50)->setCurPage(1);
```

Comment:
1. AFAIK current page is set to 1 by default, no need to specify it
2. Could you extract `50` to system configuration, near `sales_email/general/async_sending` and make it configurable via admin? Please make it configurable globally only (the same as `sales_email/general/async_sending`)